src/main/java/org/mitre/hapifhir/SubscriptionInterceptor.java

```diff
@@ -0,0 +1,275 @@
+        this.jparser = this.myCtx.newJsonParser();
+    }
+
+    /**
+     * Hook for server incoming request post processed pointcut. This handles
+     * setting a requested subscription status to active. 
+     * 
+     * @param theRequestDetails - HAPI interceptor request details
+     */
+    @Hook(Pointcut.SERVER_INCOMING_REQUEST_POST_PROCESSED)
+    public void processSubscriptions(RequestDetails theRequestDetails) {
+        String resourceName = theRequestDetails.getResourceName();
+        if (resourceName != null && resourceName.equals("Subscription")) {
+            RequestTypeEnum requestType = theRequestDetails.getRequestType();
+            if (requestType.equals(RequestTypeEnum.POST) || requestType.equals(RequestTypeEnum.PUT)) {
+                try {
+                    Subscription subscription = 
+                        this.jparser.parseResource(Subscription.class, theRequestDetails.getReader());
+                    if (subscription.getStatus().equals(SubscriptionStatus.REQUESTED)) {
+                        if (subscription.getChannel().getType().equals(SubscriptionChannelType.RESTHOOK)) {
+                            subscription.setStatus(SubscriptionStatus.ACTIVE);
+                            myLogger.info(subscription.getId() + " status set to active.");
+                        } else {
+                            subscription.setStatus(SubscriptionStatus.ERROR);
+                            myLogger.info(subscription.getId() 
+                                + " requested with invalid channel. Currently only rest hook supported.");
+                        }
+                    }
+
+                    // The line above which parses the resource consumes the input strean so we must
+                    // reset it again for handlers down the line
+                    String newInputStream = this.jparser.encodeResourceToString(subscription);
+                    theRequestDetails.setRequestContents(newInputStream.getBytes());
+                } catch (DataFormatException | IOException e) {
+                    myLogger.error("Error reading requested Subscription from stream", e);
+                }
+            }
+        }
+    }
+
+    /**
+     * Hook for server outgoing response pointcut. This handles checking
+     * if any subscriptions need to be notified. At this pointcut the 
+     * resource is accessible and all HAPI processing is complete.
+     * 
+     * @param theRequestDetails - HAPI interceptor request details
+     * @param theResource - the resource being returned by the request
+     * @return true when complete
+     */
+    @Hook(Pointcut.SERVER_OUTGOING_RESPONSE)
+    public boolean outgoingResponse(RequestDetails theRequestDetails, IBaseResource theResource) {
+        // Determine which SubscriptionTopics, if any, should be triggered
+        List<SubscriptionTopic> matchedSubscriptionTopics = getSubscriptionTopics(theRequestDetails, theResource);
```

Comment:
Thoughts on filtering down to only POST/PUT/DELETE requests at the top? We know we will never send a notification on a GET so we can save some processing time by exiting early and not searching over the whole list of subscriptions. There's probably even more filtering we could do right off the bat